app/Composers/StatusPageComposer.php

```diff
@@ -62,9 +64,100 @@ public function compose(View $view)
+
+
+        // Last week
+        $last7Days = array();
+        $last7Days[0] = date("Y-m-d", strtotime("7 days ago"));
+        $last7Days[1] = $endOfDay;
+        $downtimeCollectionLastWeek = Downtime::whereBetween('created_at', [$last7Days[0], $last7Days[1]])
+                                                ->orWhereBetween('resolved_at', [$last7Days[0], $last7Days[1]])
+                                                ->get();
+        array_push($availability, $this->calculateAvailability($downtimeCollectionLastWeek, $last7Days));
+
+        // Last month
+        $last30Days = array();
+        $last30Days[0] = date("Y-m-d", strtotime("30 days ago"));
+        $last30Days[1] = $endOfDay;
+        $downtimeCollectionLastMonth = Downtime::whereBetween('created_at', [$last30Days[0], $last30Days[1]])
+                                                ->orWhereIn('resolved_at', [$last30Days[0], $last30Days[1]])
+                                                ->get();
+
+        array_push($availability, $this->calculateAvailability($downtimeCollectionLastMonth, $last30Days));
+
+        // Current year
+        $currentYear = array();
+        $currentYear[0] = date("Y-m-d", strtotime("first day of january this year"));
+        $currentYear[1] = $endOfDay;
+        $downtimeCollectionCurrentYear = Downtime::where('created_at', "like", date("Y")."-%")
+                                                ->orWhere('resolved_at', "like", date("Y")."-%")
+                                                ->get();
+        array_push($availability, $this->calculateAvailability($downtimeCollectionCurrentYear, $currentYear));
+
+        // Last year
+        $lastYear = array();
+        $lastYear[0] = date("Y-m-d", strtotime("first day of january last year"));
+        $lastYear[1] = date("Y-m-d", strtotime("last day of december last year"));
+        $downtimeCollectionLastYear = Downtime::where('created_at', "like", date("Y", strtotime("-1 year"))."-%")
+                                                ->orWhere('resolved_at', "like", date("Y", strtotime("-1 year"))."-%")
+                                                ->get();
+        array_push($availability, $this->calculateAvailability($downtimeCollectionLastYear, $lastYear));
+
+
         $view->with($status)
             ->withComponentGroups($componentGroups)
             ->withUngroupedComponents($ungroupedComponents)
-            ->withScheduledMaintenance($scheduledMaintenance);
+            ->withScheduledMaintenance($scheduledMaintenance)
+            ->withUptimeStats($availability);
+    }
+
+    private function calculateAvailability($_downtimeCollection, $_range)
+    {
+        $availability = 100;
+
+        $start = Carbon::parse($_range[0]);
+        $end = Carbon::parse($_range[1]);
+
+        $daysInRange = $start->diffInDays($end);
+
+
+        foreach ($_downtimeCollection as $downtime)
+        {
+            // The begin of the downtime is older than today
```

Comment:
Shouldn't it mean  "... older than start day" because $start is the first day of your range?